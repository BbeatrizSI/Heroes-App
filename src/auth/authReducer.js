import { types } from "../types/types";


// const state = {
//     name: 'BÃ¡rbara',
//     logged: true
// }

export const authReducer = (state = {}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
    
        case types.logout:
            return {
                logged: false
            }
            
        default:
            return state;
    }
};

