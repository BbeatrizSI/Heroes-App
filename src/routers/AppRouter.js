import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from "react-router-dom";
import LoginScreen from '../components/login/LoginScreen';
import { Navbar } from '../components/ui/NavBar';

const AppRouter = () => {
    return (
        <Router>
        <div>
            <Navbar />  

            <Switch>
                <Route exact path="/login" component={ LoginScreen } />
                <Route exact path="/" component={ } />
            </Switch>
        </div>
      </Router>
    );
};

export default AppRouter;