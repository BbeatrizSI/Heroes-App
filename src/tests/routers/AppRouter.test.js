import React from 'react';
import { mount } from 'enzyme';
import AppRouter from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('Test in <AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('should show login if not authenticated', () => {
        
        const wrapper = mount(
                <AuthContext.Provider value= { contextValue }>
                    <AppRouter />
                </AuthContext.Provider>
            
        );

        expect( wrapper).toMatchSnapshot();
        
    });

    test('should show Marvel if is authenticated', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Lara'
            }
        }
        
        const wrapper = mount(
                <AuthContext.Provider value= { contextValue }>
                    <AppRouter />
                </AuthContext.Provider>
            
        );

        expect( wrapper.find('.navbar').exists()).toBe(true);
        
    });
    
})