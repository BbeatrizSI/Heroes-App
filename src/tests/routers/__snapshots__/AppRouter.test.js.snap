// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter /> should show login if not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
